src/ui/common/src/reducers/integration.ts

```diff
@@ -225,10 +229,110 @@ export const handleTestConnectIntegration = createAsyncThunk<
   }
 );
 
+export const handleConnectToNewIntegration = createAsyncThunk<
+  void,
+  {
+    apiKey: string;
+    service: Service;
+    name: string;
+    config: IntegrationConfig;
+  }
+>(
+  'integration/connect',
+  async (
+    args: {
+      apiKey: string;
+      service: Service;
+      name: string;
+      config: IntegrationConfig;
+    },
+    thunkAPI
+  ) => {
+    const { apiKey, service, name, config } = args;
+    Object.keys(config).forEach((k) => {
+      if (config[k] === undefined) {
```

Comment:
is config[k] allowed to be null? I'm not sure what all values are allowed for config keys, but might also be worth trying the truthy check: 
```
if (!config[k])
```

Again, depends on the values that config[k] is allowed to be. (not 100% sure here off the top of my head)

Reply:
This code is copied from somewhere but I think what you suggested is correct